src/Psalm/Internal/Analyzer/Statements/Expression/Call/ArgumentAnalyzer.php

```diff
@@ -1628,4 +1635,34 @@ private static function processTaintedness(
 
         return $input_type;
     }
+
+    private static function extendKeyedArrayComparison(Union $param_type, Union $input_type): string
+    {
+        // Default value is an empty string
+        $extension = '';
+
+        // Not sure if there's a better or more robust way to do this
+        $param_types = $param_type->getAtomicTypes();
```

Comment:
yeah, something like that:

```php
if ($param_type->isSingle()) {
    $first_param_type = $param_type->getSingleAtomic();
}
```

Then you don't need the isset on ['array'] and you can go on with the instanceof.

You may not want to try handling the Union when there is anything else in it (except maybe null), because you risk having weird results like

`... expects array{...very long list of types...}|string , possibly different type  array{...exact same list...}|int provided. The differences are in the following keys: (nothing)`

Reply:
> You may not want to try handling the Union when there is anything else in it

@orklah That's what I said :stuck_out_tongue:

Actually, I think if the containing Union has other stuff it's probably fine, but this message should probably only be generated if the compared union is a single atomic.